### PR TITLE
Add output-json option

### LINKED ISSUE #20: Command line option for output format
It would be great to be able to specify the JSON reporter format as a command-line option, e.g.
```
xcprofiler MyApp --output-json MyApp-xcprofile.json
```
I think this would allow for easier integration with continuous integration scripts.

Thanks for an awesome project!

### PR DESCRIPTION
**Pull request by giginet**
closes #20 

Add command line option `--output-path`. You can specify the path indicates output path for `JSONReporter`.

```console
$ xcprofiler --output-path "result.json" MyApp
```

### CODE REVIEW

File: lib/xcprofiler.rb [hunk 1]
```diff
@@ -28,6 +28,7 @@ def execute(args)
         opts.on("--derived-data-path", String, "Root path of DerivedData") { |v| options.derived_data_path = v }
         opts.on("-t", "--truncate-at [TRUNCATE_AT]", Integer, "Truncate the method name with specified length") { |v| options.truncate_at = v }
         opts.on("--[no-]unique", "Reject duplicated location results or not") { |v| options.unique = v }
+        opts.on("--output-path [PATH]", String, "File path to output reporters' result") { |v| options.output_path = v }
```
Comment (timakin) [inline lib/xcprofiler.rb]: Should be -o and --output.
At least -path is needless.

File: lib/xcprofiler.rb [hunk 2]
```diff
@@ -50,12 +51,23 @@ def execute(args)
...
+            order: order,
+            show_invalid_locations: options[:show_invalid_locations],
+            truncate_at: options[:truncate_at],
+            unique: options[:unique]
+        }
+        reporters = [StandardOutputReporter.new(reporter_args)]
+        if options[:output_path]
+          json_args = options.dup
+          json_args[:output_path] = options[:output_path]
+          reporters << JSONReporter.new(json_args)
```
Comment (timakin) [inline lib/xcprofiler.rb]: No other style? Only JSON?

**Comment by timakin** [comment]
Is there no test? Maybe you could add the test that generates temporary output JSON and has a validation of regex-match.

File: lib/xcprofiler.rb (hunk 1 above)
Comment (giginet) [reply lib/xcprofiler.rb]: `-o` is already used. So I renamed to `--output`

File: lib/xcprofiler.rb (hunk 2 above)
Comment (giginet) [reply lib/xcprofiler.rb]: Currently, only JSON

**Comment by giginet** [comment]
There are no tests parsing options. I'm going to add them later.

**Comment by giginet** [comment]
@timakin I fixed.

**Comment by timakin** [comment]
LGTM

**Comment by giginet** [comment]
:heart:

**Pull request merged**